backend/alert/models.py

```diff
@@ -279,28 +314,99 @@ def save(self, *args, **kwargs):
                 self.original_created_at = self.get_original_registration_rec().created_at
         super().save()
 
+    @staticmethod
+    def is_active_filter():
+        """
+        Returns a dict of filters defining the behavior of the is_active property.
+        Also used in database filtering of registrations (you cannot filter by a property value).
+        """
+        return {"notification_sent": False, "deleted": False, "cancelled": False}
+
     @property
     def is_active(self):
         """
         True if the registration would send an alert hen the watched section changes to open,
         False otherwise. This is equivalent to not(notification_sent or deleted or cancelled).
         """
-        return not (self.notification_sent or self.deleted or self.cancelled)
-
-    def alert(self, forced=False, sent_by=""):
-        if forced or self.is_active:
-            text_result = Text(self).send_alert()
-            email_result = Email(self).send_alert()
-            logging.debug("NOTIFICATION SENT FOR " + self.__str__())
-            self.notification_sent = True
-            self.notification_sent_at = timezone.now()
-            self.notification_sent_by = sent_by
-            self.save()
-            if self.auto_resubscribe:
-                self.resubscribe()
-            return (
-                email_result is not None and text_result is not None
-            )  # True if no error in email/text.
+        for k, v in self.is_active_filter().items():
+            if getattr(self, k) != v:
+                return False
+        return True
+
+    @staticmethod
+    def is_waiting_for_close_filter():
+        """
+        Returns a dict of filters defining the behavior of the is_waiting_for_close property.
+        Also used in database filtering of registrations (you cannot filter by a property value).
+        """
+        return {
+            "notification_sent": True,
+            "close_notification": True,
+            "deleted": False,
+            "cancelled": False,
+            "close_notification_sent": False,
+        }
+
+    @property
+    def is_waiting_for_close(self):
+        """
+        Returns True iff the registration would send a close notification
+        when the watched section changes to closed
+        """
+        for k, v in self.is_waiting_for_close_filter().items():
```

Comment:
if this function is the only place the above dict is used, should probably move this inline.

Reply:
It is also used in alert/tasks.py (line 47), to filter registrations (since you can't filter on a property):

`return list(section.registration_set.filter(**Registration.is_waiting_for_close_filter()))`